File: Shared/Models/EmployeeTask.cs
using System;
using Shared.Enums;

namespace Shared.Models
{
  public class EmployeeTask : BaseModel
  {
    // ======================================= //
    public TimeSpan Estimate { get; set; }
    public TaskPriority Priority { get; set; }
    public TaskState State { get; set; }
    // ======================================= //
    //        Navigation Properties
    // ======================================= //
    public int? EmployeeId { get; set; }
    public virtual Employee Employee { get; set; }
    // ======================================= //
    public EmployeeTask() : base() { }
    public EmployeeTask(string title, string description, int employeeId, TimeSpan estimate, TaskPriority priority = TaskPriority.Low, TaskState state = TaskState.New) : base(title, description)
    {
      this.Estimate = estimate;
      this.Priority = priority;
      this.State = state;
      this.EmployeeId = employeeId;
    }
    public EmployeeTask(int id, string title, string description, int employeeId, TimeSpan estimate, TaskPriority priority = TaskPriority.Low, TaskState state = TaskState.New) : this(title, description, employeeId, estimate, priority, state)
    {
      this.Id = id;
    }
  }
}

File: Client/src/app/@common/models/employee-task.ts
import { Time         } from '@angular/common'          ;
import { TaskPriority } from '@enums/task-priority.enum';
import { TaskState    } from '@enums/task-state.enum'   ;
import { BaseModel    } from '@models/base-model'       ;

export class EmployeeTask extends BaseModel {
  // ======================================= //
  public estimate  : Time        ;
  public priority  : TaskPriority;
  public state     : TaskState   ;
  public employeeId: number      ;
  // ======================================= //
  constructor(id: number, description: string, title: string, created: Date, estimate: Time, priority: TaskPriority, state: TaskState, employeeId: number) {
    super(id, description, title, created);
    this.estimate   = estimate  ;
    this.priority   = priority  ;
    this.state      = state     ;
    this.employeeId = employeeId;
  }
  // ======================================= //
}
